workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,155 @@
+using System.Linq;
+using Unity.Collections;
+using Unity.Entities;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+// ReSharper disable ClassNeverInstantiated.Global
+
+#endregion
+
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectWorldCommandSystem : ComponentSystem
+    {
+        private struct ReserveEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<Commands.WorldCommands.ReserveEntityIds.CommandResponses> Responses;
+        }
+
+        [Inject] private ReserveEntityResponseData reserveEntityResponseData;
+
+        private struct CreateEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<Commands.WorldCommands.CreateEntity.CommandResponses> Responses;
+        }
+
+        [Inject] private CreateEntityResponseData createEntityResponseData;
+
+        private struct DeleteEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<Commands.WorldCommands.DeleteEntity.CommandResponses> Responses;
+        }
+
+        [Inject] private DeleteEntityResponseData deleteEntityResponseData;
+
+        [Inject] private GameObjectDispatcherSystem GameObjectDispatcherSystem;
+
+        private static readonly InjectableId WorldCommandResponseHandlerInjectableId =
+            new InjectableId(InjectableType.WorldCommandResponseHandler, InjectableId.NullComponentId);
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < reserveEntityResponseData.Length; ++i)
+            {
+                if (reserveEntityResponseData.Responses[i].Responses.Count == 0)
+                {
+                    continue;
+                }
+
+                var entity = reserveEntityResponseData.Entities[i];
+
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
```

Comment:
This made more sense as a component to add onto an entity. This is fine if it's how everything else does it, but eventually we should replace all these with something that works with an ECS.

Reply:
I agree, unfortunately need to jump through many hoops and create storage handlers and do a triple backflip to do that with the restrictions we have at the moment

Reply:
Which restrictions are that out of interest?

Reply:
These are the two main ones for me

- Cannot have lists in IComponentData
- Cannot have non-pure components on entities

Reply:
Additionally the needed data is already present somewhere and it would be duplicated. When in a polish pass we move them as part of entities, this hack will go away too.

Reply:
We have lots of lists on IComponentData. But there is also BufferArray for that. What is a non-pure component in this context?

Reply:
It is definitely possible to implement this. We don't have to do it now. But it can definitely be done. In fact you did it for world commands the previous PR.

Reply:
Yeah I think that duplication of functionality is the biggest reason to just go like this for now, relying on the InjectableStore. We can later see about putting those on the entities more nicely

Reply:
That's partially why it was a huge PR - the handlers/responders would need to reference other systems so they would need to either add more entities or register themselves by going through the world, then the system or handlers need to manage these components' storage and lifecycle,

Reply:
I think duplication is a lesser sin than making it harder to read and slower. But for now it doesn't matter too much.

Reply:
UTY-1146